modules/gui.py

```diff
@@ -4444,6 +4445,63 @@ def setter_extra(_=None):
             imgui.end_table()
             imgui.spacing()
 
+        if draw_settings_section("Proxy"):
+            draw_settings_label(
+                "Type:",
+                "** Restart to apply changes **\n\nProxies with authentication aren't supported in login "
+                "dialog (SOCKS4 is not supported by Qt at all, SOCKS5 with authentication won't work and HTTP proxy "
+                "authentication just not implemented"
+            )
+            if imgui.begin_combo(f"###proxy_type", set.proxy_type.display):
+                for proxy_type in ProxyType:
+                    selected = proxy_type is set.proxy_type
+                    pos = imgui.get_cursor_pos()
+                    if imgui.selectable(f"###proxy_type_{proxy_type.name}", selected)[0]:
+                        set.proxy_type = proxy_type
+                        async_thread.run(db.update_settings("proxy_type"))
+                    if selected:
+                        imgui.set_item_default_focus()
+                    imgui.set_cursor_pos(pos)
+                    # self.draw_status_widget(proxy_type)
+                    imgui.same_line()
+                    imgui.text(proxy_type.display)
+                imgui.end_combo()
+
+            if set.proxy_type != ProxyType.none:
+                draw_settings_label("Address:", "** Restart to apply changes **\n\nDomain or IP address")
+                changed, value = imgui.input_text("###proxy_address", set.proxy_address)
+                if changed:
+                    set.proxy_address = value
+                    async_thread.run(db.update_settings("proxy_address"))
+
+                draw_settings_label("Port:", "** Restart to apply changes **")
+                changed, value = imgui.drag_int("###proxy_port", set.proxy_port, change_speed=0.5, min_value=1, max_value=65535)
+                set.proxy_port = min(max(value, 1), 65535)
+                if changed:
+                    set.proxy_port = int(value)
+                    async_thread.run(db.update_settings("proxy_port"))
+
+                draw_settings_label("Username:", "** Restart to apply changes **\n\nLeave empty proxy does not require authentication")
+                changed, value = imgui.input_text("###proxy_username", set.proxy_username)
+                if changed:
+                    set.proxy_username = value
+                    async_thread.run(db.update_settings("proxy_username"))
+
+                draw_settings_label("Password:", "** Restart to apply changes **\n\nLeave empty proxy does not require authentication")
+                changed, value = imgui.input_text(
+                    "###proxy_password",
+                    set.proxy_password,
+                    flags=imgui.INPUT_TEXT_PASSWORD,
+                )
+                if changed:
+                    set.proxy_password = value
+                    async_thread.run(db.update_settings("proxy_password"))
```

Comment:
prefer to show the fields, but disabled, rather than hiding.

also could use some examples, like does "address" include the http:// schema? if not, maybe it should be called host instead. and examples in the tooltip would make this clearer too